In [518]:
# imports relevant modules/packages
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import MinMaxScaler
from sklearn.preprocessing import StandardScaler
from sklearn.metrics import accuracy_score

Read diabetes.csv

In [519]:
diabetes_df = pd.read_csv('diabetes.csv') # Loads the dataset
diabetes_df.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

Differentiate between the independent variables and the dependent variable and assign them to variables x and y.

In [520]:
# independent
x = diabetes_df.drop(['Outcome'], axis=1)
# dependent
y = diabetes_df.Outcome

Generate training and test sets comprising 80% and 20% of the data respectively.

In [521]:
x_train,x_test,y_train,y_test = train_test_split(x,y,test_size = 0.20, shuffle=True, stratify=y)

print ("Training data:",x_train.shape, y_train.shape)
print ("Test data:",x_test.shape, y_test.shape)

Training data: (614, 8) (614,)
Test data: (154, 8) (154,)


768 data points split into an 80:20 ratio of 614 training observations with 8 columns and 154 test observations.

Use a MinMaxScaler and StandardScaler from sklearn.preprocessing. Fit these scalers on the train set and use these fit scalers to transform the train and test sets.

In [522]:
# define scaler
scaler_minmax = MinMaxScaler()
# fit minmaxscaler on train sets
min_max = scaler_minmax.fit(x_train)
# use fit scaler to transform both sets
min_max.transform(x_train)
min_max.transform(x_test)

# define scaler
scaler_std = StandardScaler()
# fit stdscaler on train sets
std = scaler_std.fit(x_train)
# use fit scaler to transform both sets
std.transform(x_train)
std.transform(x_test)

array([[-0.85099486, -1.23895238, -0.24276622, ..., -1.33611247,
        -0.16787455, -0.88331384],
       [ 0.33182539, -0.51646464,  0.16715736, ...,  0.64299861,
        -0.95541092, -0.45680024],
       [ 0.92323551,  1.17981091, -0.24276622, ..., -0.55455268,
        -0.54010854,  0.56683241],
       ...,
       [ 0.33182539, -0.3908146 ,  0.3208787 , ...,  0.52954638,
         0.23512257,  2.27288682],
       [-0.5552898 , -0.29657707,  0.47460004, ...,  0.95814369,
        -0.90618989, -0.79801112],
       [-0.25958474, -1.17612736, -0.03780443, ...,  0.01270846,
         0.3735567 , -0.7127084 ]])

Generate a multiple linear regression model using the training set. Use all of the independent variables.

In [523]:
# create an empty linear regression model
model_train = LinearRegression()

# to create and fit model to training sets
model_train.fit(x_train,y_train)

LinearRegression()

Print out the intercept and coefficients of the trained model.

In [524]:
# prints wanted info
print('Intercept: \n', model_train.intercept_)
print('Coefficients: \n', model_train.coef_)

Intercept: 
 -0.8902748243243805
Coefficients: 
 [ 0.01973273  0.00592094 -0.00225274  0.00020191 -0.00030388  0.01403553
  0.1241328   0.00339786]


Generate predictions for the test set.

In [525]:
# make predictions
predictions = model_train.predict(x_test)
predicted_values = predictions[0:10]
predicted_values

array([-0.1282261 ,  0.20777078,  0.60209019,  0.5842146 ,  0.12543957,
        0.629783  ,  0.32219647,  0.13024789,  0.5739677 ,  0.35595877])

For these predictions, set all predicted values below 0.5 to 0 and all predicted values of 0.5 or above to 1.

In [526]:
# creates an empty list for the adjusted predictions set to 0 and 1
adj_predictions = []
# iterates through all predictions
for i in range(len(predicted_values)):
    # set predictions below 0.5 to 0
    if predicted_values[i] < 0.5:
        adj_predictions.append(0)
    # set all predictions of 0.5 and above to 1
    else:
        adj_predictions.append(1)
adj_predictions # new predicted values

[0, 0, 1, 1, 0, 1, 0, 0, 1, 0]

Compute the accuracy_score of your model on the test set. You can get the accuracy_score from sklearn.metrics.

In [527]:
# computes accuracy score on test set
y_true = y_test.iloc[0:10] # defines y_true as values from test set
accuracy_score(y_true, adj_predictions)

0.8